### PR TITLE
chicken: bump version

### PR DESCRIPTION
**Pull request by eniehack**
bumped to 5.2.0.

### CODE REVIEW

**Comment by extrowerk** [comment]
Build fails here on 64 bit:
```
waiting for build package chicken-5.2.0-1 to be activated
waiting for build package chicken_devel-5.2.0-1 to be activated
Building ...
echo '#define C_CHICKEN_PROGRAM "chicken"' >> chicken-defaults.h
chicken chicken.read-syntax.import.scm -optimize-level 2 -include-path . -include-path ./ -inline -ignore-repository -feature chicken-bootstrap -no-warnings -specialize -consult-types-file ./types.db  -feature chicken-compile-shared -dynamic -no-trace -output-file chicken.read-syntax.import.c
./
chicken chicken.repl.import.scm -optimize-level 2 -include-path . -include-path ./ -inline -ignore-repository -feature chicken-bootstrap -no-warnings -specialize -consult-types-file ./types.db  -feature chicken-compile-shared -dynamic -no-trace -output-file chicken.repl.import.c
make: chicken: Command not found
make: chicken: Command not found
rules.make:871: recipe for target 'chicken.read-syntax.import.c' failed
make: *** [chicken.read-syntax.import.c] Error 127
make: *** Waiting for unfinished jobs....
rules.make:871: recipe for target 'chicken.repl.import.c' failed
make: *** [chicken.repl.import.c] Error 127
echo '#ifndef C_INSTALL_CC' >> chicken-defaults.h
echo '# define C_INSTALL_CC "gcc"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_INSTALL_CXX' >> chicken-defaults.h
echo '# define C_INSTALL_CXX "g++"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_INSTALL_POSTINSTALL_PROGRAM' >> chicken-defaults.h
echo '# define C_INSTALL_POSTINSTALL_PROGRAM "true"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_INSTALL_RC_COMPILER' >> chicken-defaults.h
echo '# define C_INSTALL_RC_COMPILER ""' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_INSTALL_CFLAGS' >> chicken-defaults.h
echo '# define C_INSTALL_CFLAGS "-fno-strict-aliasing -fwrapv -DHAVE_CHICKEN_CONFIG_H -DC_ENABLE_PTABLES -Os -fomit-frame-pointer"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_INSTALL_LDFLAGS' >> chicken-defaults.h
echo '# define C_INSTALL_LDFLAGS " "' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_INSTALL_PREFIX' >> chicken-defaults.h
echo '# define C_INSTALL_PREFIX "/packages/chicken-5.2.0-1/.self"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_INSTALL_SHARE_HOME' >> chicken-defaults.h
echo '# define C_INSTALL_SHARE_HOME "/packages/chicken-5.2.0-1/.self/data/chicken"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_INSTALL_BIN_HOME' >> chicken-defaults.h
echo '# define C_INSTALL_BIN_HOME "/packages/chicken-5.2.0-1/.self/bin"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_INSTALL_EGG_HOME' >> chicken-defaults.h
echo '# define C_INSTALL_EGG_HOME "/packages/chicken-5.2.0-1/.self/non-packaged/lib/chicken/11"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_INSTALL_LIB_HOME' >> chicken-defaults.h
echo '# define C_INSTALL_LIB_HOME "/packages/chicken-5.2.0-1/.self/lib"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_INSTALL_LIB_NAME' >> chicken-defaults.h
echo '# define C_INSTALL_LIB_NAME "chicken"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_INSTALL_STATIC_LIB_HOME' >> chicken-defaults.h
echo '# define C_INSTALL_STATIC_LIB_HOME "/packages/chicken-5.2.0-1/.self/lib"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_INSTALL_INCLUDE_HOME' >> chicken-defaults.h
echo '# define C_INSTALL_INCLUDE_HOME "/packages/chicken-5.2.0-1/.self/develop/headers/chicken"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_INSTALL_MORE_LIBS' >> chicken-defaults.h
echo '# define C_INSTALL_MORE_LIBS "-lnetwork"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_INSTALL_MORE_STATIC_LIBS' >> chicken-defaults.h
echo '# define C_INSTALL_MORE_STATIC_LIBS "-lnetwork"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_STACK_GROWS_DOWNWARD' >> chicken-defaults.h
echo '# define C_STACK_GROWS_DOWNWARD 1' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_TARGET_MORE_LIBS' >> chicken-defaults.h
echo '# define C_TARGET_MORE_LIBS "-lnetwork"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_TARGET_MORE_STATIC_LIBS' >> chicken-defaults.h
echo '# define C_TARGET_MORE_STATIC_LIBS "-lnetwork"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_TARGET_CC' >> chicken-defaults.h
echo '# define C_TARGET_CC "gcc"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_TARGET_CXX' >> chicken-defaults.h
echo '# define C_TARGET_CXX "g++"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_TARGET_RC_COMPILER' >> chicken-defaults.h
echo '# define C_TARGET_RC_COMPILER ""' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_TARGET_LIBRARIAN' >> chicken-defaults.h
echo '# define C_TARGET_LIBRARIAN "ar"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_TARGET_INSTALL_PROGRAM' >> chicken-defaults.h
echo '# define C_TARGET_INSTALL_PROGRAM "install"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_TARGET_CFLAGS' >> chicken-defaults.h
echo '# define C_TARGET_CFLAGS "-fno-strict-aliasing -fwrapv -DHAVE_CHICKEN_CONFIG_H -DC_ENABLE_PTABLES -Os -fomit-frame-pointer"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_TARGET_LIBRARIAN_FLAGS' >> chicken-defaults.h
echo '# define C_TARGET_LIBRARIAN_FLAGS "cru"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_TARGET_LDFLAGS' >> chicken-defaults.h
echo '# define C_TARGET_LDFLAGS " "' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_TARGET_INSTALL_PROGRAM_EXECUTABLE_OPTIONS' >> chicken-defaults.h
echo '# define C_TARGET_INSTALL_PROGRAM_EXECUTABLE_OPTIONS "-m 755"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_TARGET_INSTALL_PROGRAM_FILE_OPTIONS' >> chicken-defaults.h
echo '# define C_TARGET_INSTALL_PROGRAM_FILE_OPTIONS "-m 644"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_TARGET_FEATURES' >> chicken-defaults.h
echo '# define C_TARGET_FEATURES ""' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_CROSS_CHICKEN' >> chicken-defaults.h
echo '# define C_CROSS_CHICKEN 0' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_TARGET_PREFIX' >> chicken-defaults.h
echo '# define C_TARGET_PREFIX "/packages/chicken-5.2.0-1/.self"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_TARGET_BIN_HOME' >> chicken-defaults.h
echo '# define C_TARGET_BIN_HOME "/packages/chicken-5.2.0-1/.self/bin"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_TARGET_LIB_HOME' >> chicken-defaults.h
echo '# define C_TARGET_LIB_HOME "/packages/chicken-5.2.0-1/.self/lib"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_TARGET_LIB_NAME' >> chicken-defaults.h
echo '# define C_TARGET_LIB_NAME "chicken"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_TARGET_RUN_LIB_HOME' >> chicken-defaults.h
echo '# define C_TARGET_RUN_LIB_HOME "/packages/chicken-5.2.0-1/.self/lib"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_TARGET_SHARE_HOME' >> chicken-defaults.h
echo '# define C_TARGET_SHARE_HOME "/packages/chicken-5.2.0-1/.self/share"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_TARGET_INCLUDE_HOME' >> chicken-defaults.h
echo '# define C_TARGET_INCLUDE_HOME "/packages/chicken-5.2.0-1/.self/include/chicken"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_TARGET_STATIC_LIB_HOME' >> chicken-defaults.h
echo '# define C_TARGET_STATIC_LIB_HOME "/packages/chicken-5.2.0-1/.self/lib"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_CHICKEN_PROGRAM' >> chicken-defaults.h
echo '# define C_CHICKEN_PROGRAM "chicken"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_CSC_PROGRAM' >> chicken-defaults.h
echo '# define C_CSC_PROGRAM "csc"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_CSI_PROGRAM' >> chicken-defaults.h
echo '# define C_CSI_PROGRAM "csi"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_CHICKEN_DO_PROGRAM' >> chicken-defaults.h
echo '# define C_CHICKEN_DO_PROGRAM "chicken-do"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_CHICKEN_INSTALL_PROGRAM' >> chicken-defaults.h
echo '# define C_CHICKEN_INSTALL_PROGRAM "chicken-install"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_CHICKEN_UNINSTALL_PROGRAM' >> chicken-defaults.h
echo '# define C_CHICKEN_UNINSTALL_PROGRAM "chicken-uninstall"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_CHICKEN_STATUS_PROGRAM' >> chicken-defaults.h
echo '# define C_CHICKEN_STATUS_PROGRAM "chicken-status"' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_WINDOWS_SHELL' >> chicken-defaults.h
echo '# define C_WINDOWS_SHELL 0' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_BINARY_VERSION' >> chicken-defaults.h
echo '# define C_BINARY_VERSION 11' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '#ifndef C_USES_SONAME' >> chicken-defaults.h
echo '# define C_USES_SONAME 1' >> chicken-defaults.h
echo '#endif' >> chicken-defaults.h
echo '/* END OF FILE */' >> chicken-defaults.h
Warning: Command '['bash', '-c', '. /wrapper-script']' returned non-zero exit status 2.
Error: Build has failed - stopping.
/Ports/haikuports> 
```

**Comment by extrowerk** [comment]
Do i see it correctly chicken requires chicken to build?

**Comment by korli** [comment]
bootstrapping section: https://github.com/kdltr/chicken-core/blob/master/README#L650

**Comment by eniehack** [comment]
ChickenScheme not supported pararallel builds, so I deleted `$jobArgs` in `BUILD()`.

>  Note that parallel builds (using the "-j" make(1) option) are *not* supported.
https://code.call-cc.org/cgi-bin/gitweb.cgi?p=chicken-core.git;a=blob;f=README;h=3239560a2c30d6959bc2822689324ab77ecb4419;hb=69d4574dc32b398e9fbe34cef40f144194997237#l84

File: dev-scheme/chicken/chicken-5.2.0.recipe [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+CHICKEN runs on Linux, MacOS X, Windows, and many Unix flavours."
+HOMEPAGE="http://www.call-cc.org/"
+COPYRIGHT="2008-2020 The Chicken Team
+	2000-2007 Felix L. Winkelmann"
+LICENSE="BSD (2-clause)"
+REVISION="1"
+SOURCE_URI="http://code.call-cc.org/releases/$portVersion/chicken-$portVersion.tar.gz"
+CHECKSUM_SHA256="819149c8ce7303a9b381d3fdc1d5765c5f9ac4dee6f627d1652f47966a8780fa"
+
+ARCHITECTURES="?x86_gcc2 ?x86 x86_64"
```
Comment (Begasus) [inline dev-scheme/chicken/chicken-5.2.0.recipe:14]: For now you can strike x86_gcc2 as a possible candidate, too much parse errors :)
```diff
- ARCHITECTURES="?x86_gcc2 ?x86 x86_64"
+ ARCHITECTURES="!x86_gcc2 ?x86 x86_64"
```

**Comment by Begasus** [comment]
Builds OK on R1B2 (32bit) for secondary architecture, maybe you could remove the "old" recipes while in the process?

**Comment by extrowerk** [comment]
Now it compiles just fine on 64 bit too.

**Comment by Begasus** [comment]
@eniehack  could you remove the "old" recipes while in the process? Then we can merge (if no one has any objections)

**Comment by Begasus** [comment]
Thanks for your commit! :+1:

**Pull request merged**